### DISCUSSION TITLE
Any data validated against JSON schema.

### DISCUSSION THREAD

**Discussion by adityasivarajmz** [foxglove/community#656] [Q&A]
**Description**
With the current structure of the JSON schemas, any data will be able to be validated against any schema. Please add `"required: "` and `"additionalProperties: "` keys to the schema.

- Version:
- Platform: JSON

**Steps To Reproduce**
Try validating any random JSON message against any of the given foxglove JSON schemas.

**Expected Behavior**
Validation should fail when validating LocationFix message against CubePrimitive schema.

**Comment by jtbandes** [comment]
Currently, Foxglove Studio does not perform validation with the JSON schemas. They are used to understand the message structures and populate auto-complete for [message path syntax](https://foxglove.dev/docs/studio/app-concepts/message-path-syntax), but the app does not actually validate the messages. As such I'm converting this issue to a discussion since performing validation would be more of a new feature in the app.

**Reply by jtbandes** [reply]
If you are just interested in updating the schemas so you can do validation yourself using another tool, please consider making a PR to the https://github.com/foxglove/schemas repo where the schemas are generated.
